File: functions/updateBudget.c
#include <stdio.h>
float income, rent, utilities, groceries, transportation, savings, expenses, spend;

float input(char type[], float var){
    printf("Monthly %s:\n", type);
    scanf("%f", &var);
    return var;
}


void percent(char type[], int amount){
    int per = amount/income *100;

    printf("Your %s is %d%% of your income.\n", type, per);
}


int main(void){
    
    printf("This is a budget calculator.\n");
    income = input("income", income);
    rent = input("rent", rent);
    utilities = input("utilities", utilities);
    groceries = input("groceries", groceries);
    transportation = input("transportation", transportation);
    savings = income * .2;
    expenses = rent + utilities + groceries + transportation;
    spend = income - expenses - savings;
    printf("You make $%.2f\n", income);
    printf("Your expenses are $%.2f\n", expenses);
    printf("Your savings are $%.2f\n", savings);
    printf("Your spending money is $%.2f\n", spend);
    
    percent("rent", rent);
    percent("utilities", utilities);
    percent("groceries", groceries);
    percent("transportation", transportation);
    percent("expenses", expenses);
    percent("savings", savings);
    return 0;
}

File: functions/updateBudget.py

def user_inputs(type):
    return float(input(f"Monthly {type}"))

def divide(type, income):
    return type/ income *100

    

income = user_inputs("income: ")
rent = user_inputs("rent:")
utilities = user_inputs("utilities:")
groceries = user_inputs("groceries:")
transportation = user_inputs("transportation:")
savings = income*0.2
expenses = rent + utilities + groceries + transportation
spend = income-int(expenses)-savings

def percent(type,amount):
    per = amount / income *100
    return(f"Your {type} is {per}% income")

print(percent("rent percentage", rent))
print(percent("utilities percentage", utilities))
print(percent("groceries percentage", groceries))
print(percent("transportation percentage", transportation))
print(percent("expenses percentage", expenses))
print(percent("spending percentage", spend))